packages/consumption/src/modules/attributes/AttributesController.ts

```diff
@@ -932,30 +1020,33 @@ export class AttributesController extends ConsumptionBaseController {
             throw CoreErrors.attributes.invalidPropertyValue(`Attribute '${id}' isn't a repository attribute.`);
         }
 
-        while (repositoryAttribute.succeededBy) {
-            const successor = await this.getLocalAttribute(repositoryAttribute.succeededBy);
-            if (!successor) {
-                throw TransportCoreErrors.general.recordNotFound(LocalAttribute, repositoryAttribute.succeededBy.toString());
-            }
-
-            repositoryAttribute = successor;
-        }
-
         const query: any = { "shareInfo.sourceAttribute": repositoryAttribute.id.toString() };
-
         if (typeof peers !== "undefined") {
             query["shareInfo.peer"] = { $in: peers.map((address) => address.toString()) };
         }
-
         if (onlyLatestVersions) {
             query["succeededBy"] = { $exists: false };
         }
 
-        const ownSharedIdentityAttributeVersions: LocalAttribute[] = await this.getLocalAttributes(query);
+        const ownSharedIdentityAttributes = await this.getLocalAttributes(query);
+        const ownSharedIdentityAttributePredecessors = await this.getSharedPredecessorsOfRepositoryAttribute(repositoryAttribute, query);
+        const ownSharedIdentityAttributeSuccessors = await this.getSharedSuccessorsOfRepositoryAttribute(repositoryAttribute, query);
+
+        const ownSharedIdentityAttributeVersions = [...ownSharedIdentityAttributeSuccessors.reverse(), ...ownSharedIdentityAttributes, ...ownSharedIdentityAttributePredecessors];
+        return ownSharedIdentityAttributeVersions;
+    }
+
+    public async getSharedPredecessorsOfRepositoryAttribute(repositoryAttribute: LocalAttribute, query?: any): Promise<LocalAttribute[]> {
+        if (typeof query !== "undefined") {
+            query = JSON.parse(JSON.stringify(query));
+        } else {
+            query = {};
+        }
```

Comment:
this can be written as `query ?= {}` or you just add {} as a default in the method params